### PR TITLE
Don't log authentication configuration when connecting to MQTT broker

### PR DESCRIPTION
**Pull request by Bravo555**
PR #2233 introduced some `info!` logging statements to MQTT connection code, so that the user could see what's happening.

However, this code printed out the contents of the AuthenticationConfig struct which contained root certificate store and a client private key if client authentication option was used.

The intent was to print out the certificate paths, but what was printed was the contents, potentially including secrets.

Fortunately it was caught quickly by @reubenmiller.

I think the biggest factor that contributed to this mistake was keeping secrets together with other safe-to-print fields, without using any marker type to denote that `ClientAuthConfig.key` is a secret, disabling it's `Debug` implementation so that it can't be accidentally printed out in the first place. Crates like [secrecy](https://docs.rs/secrecy/latest/secrecy/) can be used to replace `Display` and `Debug` impls so that sensitive information cannot be printed.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by didier-wenzek — COMMENTED** [review]
To avoid repeating this mistake, I would remove the culprit `derive(Debug)`.

```
-#[derive(Debug, Clone)]
+#[derive(Clone)]
 struct ClientAuthConfig {
     cert_chain: Vec<Certificate>,
     key: PrivateKey,
 }
 
+impl Debug for ClientAuthConfig {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        // Do not print the private key!
+        self.cert_chain.fmt(f)
+    }
+}

```

**Comment by Bravo555** [comment]
> To avoid repeating this mistake, I would remove the culprit derive(Debug).

fixed in b6a13abcb556367c71efdee5a9ad6ee04767e72d.

File: crates/common/mqtt_channel/src/config.rs [hunk 1]
```diff
@@ -103,10 +105,27 @@ impl Default for AuthenticationConfig {
...
+impl Debug for ClientAuthConfig {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("ClientAuthConfig")
+            .field("cert_chain", &self.cert_chain)
+            .finish()
+    }
+}
+
+#[derive(Debug, Clone)]
+struct PrivateKey(rustls::PrivateKey);
```
Comment (didier-wenzek) [inline crates/common/mqtt_channel/src/config.rs:123]:
```suggestion
#[derive(Clone)]
struct PrivateKey(rustls::PrivateKey);
```

**Review by didier-wenzek — APPROVED** [review]
Approved. Cargo.lock has to be updated before merge.

**Review by reubenmiller — APPROVED** [review]
Approved

The log entries look much better now:

```
Sep 13 06:47:08 ad0bf762eaeb c8y-firmware-plugin[969]: 2023-09-13T06:47:08.91651Z  INFO mqtt_channel::connection: MQTT connecting to broker: host=localhost:8883, session_name=Some("c8y-firmware-plugin")
```

**Pull request merged**